### ISSUE TITLE
Incorrect unsat on mod/div example

### ISSUE THREAD

**Issue by pjljvandelaar** [CVC4/CVC4#3765]
**Describe the bug**
On the attached file
[example1.smt2.txt](https://github.com/CVC4/CVC4/files/4214382/example1.smt2.txt)
I get unsat.
Yet sat is correct.
I tried cvc4 to proof that sat is really correct: See
[proof.smt2.txt](https://github.com/CVC4/CVC4/files/4214393/proof.smt2.txt)
(only the last step gives unknown, yet I hope it also convinces you that sat is the correct outcome).

**Command line arguments**: 
 cvc4 --lang=smt2.5 --incremental --strings-exp --fmf-fun-rlv --uf-ss-fair --no-strings-print-ascii --nl-ext-tplanes

**CVC4 version/commit**: 
This is CVC4 version 1.8-prerelease
compiled with GCC version 5.3.1 20160211
on Feb 13 2020 12:13:59

**Operating system**: Windows 10

**`configure.sh` options**
not applicable, just downloaded executable from https://cvc4.cs.stanford.edu/downloads/builds/win64-opt/unstable/cvc4-2020-02-13-win64-opt.exe

**`configure.sh` output**
not applicable, just downloaded executable from https://cvc4.cs.stanford.edu/downloads/builds/win64-opt/unstable/cvc4-2020-02-13-win64-opt.exe

Note I found this bug while trying to simplify failing cases for
https://github.com/CVC4/CVC4/issues/3697

**Comment by pjljvandelaar** [comment]
Note the file can even be simplified a little more to
```
(set-info :smt-lib-version 2.5)
(set-option :produce-models true)
(set-logic ALL)
(define-funs-rec (
 (f11((va9 Int))Int)
 (f3((v1f Int))Int)
)
( 
 (f3 (ite (= 0 va9) (- 1) (div (- 1) va9)))
  0
)
)
(assert (= 0 (f11 22)))
(check-sat)
```
Making it clear that f11 should always return the return value of f3 (which is 0).

**Comment by ajreynol** [comment]
Thanks a lot for the bug report, this was very helpful.  It is fixed in #3766.
